### DISCUSSION TITLE
How should we curate `env_medium` for soil & what is the design?

### DISCUSSION THREAD

**Discussion by mslarae13** [microbiomedata/issues#837] [General]
FOR SOIL

Currently, `env_medium` for [NMDC](https://microbiomedata.github.io/berkeley-schema-fy24/env_medium/) has a manually curated list of permissible values in `EnvMediumSoilEnum ` for [the submission portal](https://microbiomedata.github.io/berkeley-schema-fy24/env_medium/).

According to GSC, `env_medium` can be [multivalued](https://genomicsstandardsconsortium.github.io/mixs/0000014/). 
(Though, that's just in the description and not in the schema "Report the **environmental material(s)** immediately surrounding the sample")

NMDC does not allow for this.

```
name: env_medium
string_serialization: '{termLabel} {[termID]}'
domain_of:
- Biosample
range: ControlledIdentifiedTermValue
multivalued: false
```

In this discussion I'd like to determine
1. The rules / guidance NMDC will use to curate the env_medium permissible values for soil in the NMDC submission portal
     - What branches? How do we query OAK for this?
2. Should this term be multivalued?
    - allow for multi-select in the submission portal like we do for analysis_type

@turbomam @cmungall @lamccue @Natalie-Winans

**Comment by turbomam** [comment]
I'm really looking forward to a group effort on this.

The OAK query for Soil env_broad_scale is a good illustration what what can be done. It might not be as easy for env_medium and is likely to be much harder for env_local_scale.

Can we create the value set first, and save the decision about multivalued and user interface until after that?

**Comment by turbomam** [comment]
We have lots of issues (in many repos) and many Google Docs that have tried to address this in the past. Should we try to catalogue all of those here, or just take a fresh start?

**Comment by mslarae13** [comment]
Decision was made in last weeks meeting for an initial subset creation. See [ADR](https://github.com/microbiomedata/issues/pull/881), but in summary

NMDC General Query
- `env_medium` will be composed of terms that branch from environmental material [ENVO:00010483]

For **soil** environment (MIxS Extension)
- `env_medium` will be textually filtered for classes with soil or soils in their name or description.
  - Will evaluate if adding stemming is needed.

This will be refined & updates after evaluation, as aligned with @turbomam 's comment above . https://github.com/microbiomedata/issues/discussions/837#discussioncomment-10388452
